lwgenerate/azure/activity_log_without_config.tf

```diff
@@ -0,0 +1,34 @@
+terraform {
+  required_providers {
+    azuread = {
+      source  = "hashicorp/azuread"
+      version = "~> 2.16"
+    }
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "=2.91.0"
+    }
+  }
+}
+
+provider "azuread" {
+}
```

Comment:
I'm wondering whether we need this provider? Looking at our TF registry module examples it looks like we don't use this in the examples.
Granted the customer-facing docs use this approach https://docs.lacework.com/azure-compliance-and-activity-log-integrations-terraform-using-azure-cloud-shell
It looks like we actually create an `az_ad_application` with each module. Perhaps we could change this to work similarly to the GCP examples, where we create the `Service Account` _(az_ad_application)_ in the config module and then re-use this in the activity log module via the config module outputs. eg.
```
provider "google" {}

provider "lacework" {}

module "gcp_organization_config" {
  source  = "lacework/config/gcp"
  version = "~> 1.0"

  org_integration = true
  organization_id = "my-organization-id"
}

module "gcp_organization_audit_log" {
  source  = "lacework/audit-log/gcp"
  version = "~> 2.0"

  bucket_force_destroy         = true
  org_integration              = true
  use_existing_service_account = true
  service_account_name         = module.gcp_organization_config.service_account_name
  service_account_private_key  = module.gcp_organization_config.service_account_private_key
  organization_id              = "my-organization-id"
  log_bucket                   = "lacework-logging-bucket"
}
```

Reply:
We likely should get the customer-facing docs for Azure to do something similar to this also
```
provider "azurerm" {
features {}
}

module "az_config" {
source  = "lacework/config/azure"
version = "~> 1.0"

}

module "az_activity_log" {
source  = "lacework/activity-log/azure"
version = "~> 1.0"

use_existing_ad_application = true
application_id              = module.az_config.application_id
application_password        = module.az_config.application_password
service_principal_id        = module.az_config.service_principal_id
}
```
Thoughts @ipcrm, @dmurray-lacework, @afiune, @marcosgm ?

Reply:
From talking to @afiune, it seems that the example that we have in the support docs has actually been a fairly recent update that was made by @marcosgm to make it more clear to the end users what is going on in the modules, rather than hiding away the app registration creation.
I still prefer the cleaner example suggested above, however, if we feel that the more verbose route shown in the docs is the right way to go, lets stick with what we have.

Reply:
Yeah, I have made it so that if the user does not enter any config then it will generate that terraform as per the support docs, but the user can enter details for an  existing AD, Resource Group etc